nemo_text_processing/g2p/modules.py

```diff
@@ -404,39 +474,44 @@ def _normalize_dict(self, phoneme_dict_obj: dict) -> Tuple[DefaultDict[str, List
         g2p_dict = defaultdict(list)
         symbols = set()
         for word, prons in phoneme_dict_obj.items():
+            # process word
             # update word cases.
-            if self.set_graphemes_upper:
-                word_new = word.upper()
-            else:
-                word_new = word.lower()
+            word_new = set_grapheme_case(word, case=self.grapheme_case)
 
             # add grapheme symbols if `use_chars=True`.
             if self.use_chars:
-                # remove punctuations after a word.
+                # remove punctuations within a word. Punctuations can exist at the start, middle, and end of a word.
+                # TODO @xueyang: is the removal of a hyphen within a word expected, e.g. up-to-date -> uptodate?
                 word_no_punct = self.PUNCT_REGEX.sub('', word_new)
```

Comment:
```
import re
PUNCT_REGEX = re.compile(fr"[^{LATIN_CHARS_ALL}\d]")
PUNCT_REGEX.sub("", "up-to-date")
Out[11]: 'uptodate'
```

Reply:
@ekmb @redoctopus could you please confirm if it needs fixed?

Reply:
I think it should be fine since this part is just to update the symbol set without including punctuation, and doesn't affect the word in the long-term. I would definitely count the hyphen as a punctuation character over a "grapheme" character, if that makes sense.

The only issue would be if the user doesn't include the hyphen in the punctuation list, in which case the entry would be invalid; I assume this would be intended if that's a deliberate decision...

Reply:
agreed. Thanks for clarification. I will keep as-is and removed the TODO comments.